### PR TITLE
 [schema] KeyValue schema support using AUTO_CONSUME as key/value schema

### PR DESCRIPTION
**Pull request by sijie**
*Motivation*

Currently KeyValue schema doesn't support using AUTO_CONSUME.

This PR is to add this support.

This PR is based on #4836

*Changes*

- refactor a bit on Schema interface to support fetching schema info for both AutoConsumeSchema and KeyValueSchema before subscribing
- add AUTO_CONSUME support to KeyValueSchema
- add tests

### CODE REVIEW

File: pulsar-broker/src/test/java/org/apache/pulsar/client/api/SimpleSchemaTest.java [hunk 1]
```diff
@@ -252,4 +264,126 @@ private void getSchemaVersionFromMessages(boolean batching) throws Exception {
...
+             .subscribe();
+             Consumer<KeyValue<V1Data, V1Data>> c2 = pulsarClient.newConsumer(
+                 Schema.KeyValue(
+                     Schema.AVRO(V1Data.class),
+                     Schema.AVRO(V1Data.class),
+                     KeyValueEncodingType.SEPARATED))
+             .topic(topic)
+             .subscriptionName("sub2")
+             .subscriptionInitialPosition(SubscriptionInitialPosition.Earliest)
+             .subscribe();
```
Comment (codelipenghui) [inline pulsar-broker/src/test/java/org/apache/pulsar/client/api/SimpleSchemaTest.java]: Will c2 be reject by broker? default is use exclusive subscription mode.

Comment (sijie) [reply pulsar-broker/src/test/java/org/apache/pulsar/client/api/SimpleSchemaTest.java]: No because they are using different subscription names

**Comment by codelipenghui** [comment]
run Integration Tests

**Comment by jiazhai** [comment]
mark 2.4.1 for conflict #4962

**Pull request merged**